### PR TITLE
Reservation Price

### PR DESCRIPTION
**Pull request by mplsmitch**
#### **What problem does your proposal solve? Please begin with the relevant issue number. If there is no existing issue, please also describe alternative solutions you have considered.**

There are a number of providers that charge a fee to reserve vehicles in advance of initiating a rental. The current specification does support this component of rental pricing.  These fees are either charged as a flat rate per reservation, or on a per minute basis for the duration of the reservation until the rental is initiated by unlocking the vehicle. Examples and further discussion of this topic can be found in issue #401 .

#### **What is the proposal?**
This PR adds two OPTIONAL fields, one which describes reservation price as a flat rate, and one that describes the rate on a per minute basis. These fields are to be used in conjunction with `vehicle_types.json#default_reserve_time` to describe the cost of reservation. They are exclusive of each other and are not to be combined within a single pricing plan.

#### **Is this a breaking change?**
- [ ] Yes 
- [x] No
- [ ] Unsure

#### **Which files are affected by this change?**
gbfs.md: system_pricing_plans.json

### CODE REVIEW

**Comment by josee-sabourin** [comment]
I hereby call a vote on this proposal. Voting will be open for 10 full calendar days until 11:59PM UTC on Friday, January 20th.

Please vote for or against the proposal, and include the organization for which you are voting in your comment.

Please note if you can commit to implementing the proposal.

**Comment by testower** [comment]
+1 from Entur

File: gbfs.md [hunk 1]
```diff
@@ -963,6 +963,8 @@ Field Name | REQUIRED | Type | Defines
...
+\-&nbsp;`reservation_price_per_min` | OPTIONAL | Non-Negative Float | The cost, described as per minute rate, to reserve the vehicle prior to beginning a rental. This amount is charged for each minute of the vehicle reservation until the rental is initiated, or until the number of minutes defined in `vehicle_types.json#default_reserve_time` elapses, whichever comes first. When using this field, you MUST declare a value in `vehicle_types.json#default_reserve_time`. This field MUST NOT be combined in a single pricing plan with `reservation_price_flat_rate`.
+\-&nbsp;`reservation_price_flat_rate` | OPTIONAL | Non-Negative Float | The cost, described as a flat rate, to reserve the vehicle prior to beginning a rental. This amount is charged once to reserve the vehicle for the duration of the time defined by `vehicle_types.json#default_reserve_time`. When using this field, you MUST declare a value in `vehicle_types.json#default_reserve_time`.This field MUST NOT be combined in a single pricing plan with `reservation_price_per_min`.
```
Comment (bdferris-v2) [inline gbfs.md]: Should `default_reserve_time` be changed to "Conditionally REQUIRED` to go along with this change?

Also nit: space after period for ".This field MUST NOT..."

Comment (josee-sabourin) [reply gbfs.md]: Good catch - I would agree. @mplsmitch, thoughts?

Comment (mplsmitch) [reply gbfs.md]: agree- I'll update the PR

**Comment by josee-sabourin** [comment]
Voting on this PR closes in 2 calendar days. Please vote for or against the proposal, and include the organization for which you are voting in your comment. Please note if you can commit to implementing the proposal.

**Comment by rootsixtysix** [comment]
+1 from nextbike by TIER

File: gbfs.md [hunk 2]
```diff
@@ -963,6 +963,8 @@ Field Name | REQUIRED | Type | Defines
...
 \-&nbsp;`currency` | Yes | String | Currency used to pay the fare. <br /><br /> This pricing is in ISO 4217 code: http://en.wikipedia.org/wiki/ISO_4217 <br />(for example, `CAD` for Canadian dollars, `EUR` for euros, or `JPY` for Japanese yen.)
 \-&nbsp;`price` | Yes | Non-Negative Float | Fare price, in the unit specified by `currency`. <br/>*(added in v2.2)* In case of non-rate price, this field is the total price. In case of rate price, this field is the base price that is charged only once per trip (typically the price for unlocking) in addition to `per_km_pricing` and/or `per_min_pricing`.
+\-&nbsp;`reservation_price_per_min` | OPTIONAL | Non-Negative Float | The cost, described as per minute rate, to reserve the vehicle prior to beginning a rental. This amount is charged for each minute of the vehicle reservation until the rental is initiated, or until the number of minutes defined in `vehicle_types.json#default_reserve_time` elapses, whichever comes first. When using this field, you MUST declare a value in `vehicle_types.json#default_reserve_time`. This field MUST NOT be combined in a single pricing plan with `reservation_price_flat_rate`.
```
Comment (kheraankit) [inline gbfs.md:966]: Curious to hear your thoughts on not doing something similar to `per_min_pricing` which provides flexibility to do tiered pricing.

Comment (mplsmitch) [reply gbfs.md:966]: I'm not sure I understand your question. These fields set a fee to reserve a vehicle prior to beginning a trip. They can be used with either `per_min_pricing`  or `per_km_pricing`.  Regardless of which type of pricing you're using you can set your reservation fee as a flat rate, or charge per minute.

**Comment by benwedge** [comment]
+1 from Joyride

**Comment by fbouchPBSC** [comment]
+1 from PBSC

**Comment by rickbruce** [comment]
+1 from Ito

**Comment by ezmckinn** [comment]
+1 from Superpedestrian. Seems like a wise extension of the spec.

**Comment by schnuerle** [comment]
+1 from OMF staff

**Comment by josee-sabourin** [comment]
This vote has now closed, and it passes!

Votes in favor:
Entur (consumer)
Nextbike (producer)
Joyride (producer)
PBSC (producer)
Ito World (producer)
Superpedestrian (producer)
OMF (3rd party)
 
There were no votes against.
Thank you to everyone who took the time to review and to vote on this! We incorporate it into 3.0-RC, which should be ready to go in the coming week.

**Review by josee-sabourin — APPROVED** [review]

**Comment by richfab** [comment]
In the absence of producers to implement this change, it will be postponed to v3.1-RC.

Please add in a comment if you are a producer planning to implement this change in the next few weeks.

Thank you!

**Comment by richfab** [comment]
This change is delayed to the next version to allow v3.0 to become the official version.

Indeed, despite the high engagement on the vote, no producer or consumer has come forward as a First Adopter for this change since its release in March 2023.

This useful change can always be added to the spec later.

Thank you all for your continued involvement in the GBFS spec. 🙏

**Pull request merged**